GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -721,9 +721,13 @@ public void AddBuff(IBuff b)
                     // Don't need the newly added buff instance as we already have a parent who we can add stacks to.
                     RemoveBuffSlot(b);
 
-                    // Refresh the time of the parent buff and add a stack.
+                    // Refresh the time of the parent buff and adds a stack if Max Stacks wasn't reached.
                     ParentBuffs[b.Name].ResetTimeElapsed();
-                    ParentBuffs[b.Name].IncrementStackCount();
+                    if (ParentBuffs[b.Name].StackCount < ParentBuffs[b.Name].MaxStacks)
```

Comment:
I think that you can simply change the `if` to `if (IncementStackCount)` and remove the Increment... from the inside of the `if` block. The Increment part should already stop at MaxStacks and return false when nothing is added.